src/tenant_tokens.rs

```diff
@@ -0,0 +1,115 @@
+use crate::{
+    errors::* 
+};
+use serde::{Serialize, Deserialize};
+use jsonwebtoken::{encode, Header, EncodingKey};
+use time::{OffsetDateTime};
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")] 
+struct TenantTokenClaim {
+    api_key_prefix: String,
+    search_rules: serde_json::Value,
+    #[serde(with = "time::serde::timestamp::option")]
+    exp: Option<OffsetDateTime>,
+}
+
+pub fn generate_tenant_token(search_rules: serde_json::Value, api_key: &str, expires_at: Option<OffsetDateTime>) -> Result<String, Error> {
+    if api_key.len() < 8 {
+        return Err(Error::TenantTokensInvalidApiKey)
+    }
+
+    if expires_at.is_some() && OffsetDateTime::now_utc() > expires_at.unwrap() {
```

Comment:
```suggestion
    if expires_at.map_or(false, |expires_at| OffsetDateTime::now_utc() > expires_at) {
```

It's more common to see it like that in rust